extensions/vertx-web/deployment/src/main/java/io/quarkus/vertx/web/deployment/VertxWebProcessor.java

```diff
@@ -599,6 +611,17 @@ void implementInvoke(HandlerDescriptor descriptor, BeanInfo bean, MethodInfo met
         invoke.returnValue(null);
     }
 
+    private static final List<DotName> TYPES_IGNORED_FOR_REFLECTION = Arrays.asList(io.quarkus.arc.processor.DotNames.STRING,
+            DotNames.BUFFER, DotNames.JSON_ARRAY, DotNames.JSON_OBJECT);
+
+    private static void registerForReflection(Type contentType,
+            BuildProducer<ReflectiveHierarchyBuildItem> reflectiveHierarchy) {
+        if (TYPES_IGNORED_FOR_REFLECTION.contains(contentType.name())) {
+            return;
+        }
+        reflectiveHierarchy.produce(new ReflectiveHierarchyBuildItem(contentType, TYPES_IGNORED_FOR_REFLECTION::contains));
```

Comment:
As mentioned in my comment, I would use that + the default predicate.

Reply:
Fixed.